### PR TITLE
Add optimizer_kwargs, scheduler, and scheduler_kwargs parameters to Task __init__ methods.

### LINKED ISSUE #729: Add missing parameters to the `__int__` method of certain tasks.
## 🚀 Feature

Add `optimizer_kwargs`, `scheduler`, and `scheduler_kwargs` parameters to the `__init__` method of all tasks that are missing it.

### Motivation

Many tasks don't have a way passing a learning rate scheduler or even different optimizer arguments to the optimizer of their choice.

### Pitch

Add 
```
        optimizer_kwargs: Optional[Dict[str, Any]] = None,
        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
        scheduler_kwargs: Optional[Dict[str, Any]] = None,
```
to the `__init__` methods of all tasks keeping the API consistent across all tasks. Update the documentation as well.

### Additional context

I have the fix ready for this to submit as a PR.

### PR DESCRIPTION
**Pull request by karthikrangasai**
## What does this PR do?

Fixes #729 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:
